doc/tutorials/content/executor.rst

```diff
@@ -0,0 +1,101 @@
+.. _executor:
+
+Using Executors
+--------------------------------------------------
+
+Executors allows control over the properties as well as the execution
+context over which a function will run. Therefore allowing various parts
+of the code running in different contexts. In this tutorial we will learn how
+to manually create and pass executors to to a FunctorFilter.
+
+.. note::
+   Not all functions support executors. Check the Code API for more info on which
+   functions are supported.
+
+
+The code
+--------
+
+First, create a file, let's say, ``executor.cpp``, and place the following inside it:
+
+.. literalinclude:: sources/executor/executor.cpp
+   :language: cpp
+   :linenos:
+
+The explanation
+---------------
+
+Now, let's break down the code piece by piece.
+
+In the following lines, we create a Point Cloud structure, then fill the
+input cloud using `CloudGenerator`. The generator uniformly fills the
+input cloud with a point cloud having width & height as 200 (using 128 as the seed value). Each
+point is generated with x,y & z co-ordinates in between the range -20 & 20.
+
+.. literalinclude:: sources/executor/executor.cpp
+   :language: cpp
+   :lines: 8-11
+
+Then, we create two executors. The first one `simple_executor` is of type `inline_executor`
+which runs code sequentially on the CPU. The second one `parallel_executor` is of type
+`omp_executor` which uses OMP to parallelize the execution. We set the max_threads that OMP
+can utilize to 4. If unspecified explicitly by default all the threads available in the system
+will be used.
+
+.. literalinclude:: sources/executor/executor.cpp
+   :language: cpp
+   :lines: 13-15
+
+
+We then create two FunctorFilter instances. The first filter `radius_filter` is used to filter out
+any points which are at a distance greater than the specified `radius_filter` from the origin.
+The second filter `positive_y_filter` filters out all points having negative y co-ordinates.
+
+.. literalinclude:: sources/executor/executor.cpp
+   :language: cpp
+   :lines: 17-33
+
+Finally, we call `filter` and with the respective executors. For `radius_filter` we pass the
+`parallel_executor` which will spawn 4 threads and and spilt the filtering of points between them.
```

Comment:
create 4 OMP contexts

The OMP call can create 3 new threads and use the main thread as the 4th or re-use 1-4 threads from it's own pool of thread contexts

Reply:
Updated